testing/src/main/java/io/stargate/it/grpc/ExecuteBatchStreamingTest.java

```diff
@@ -0,0 +1,158 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(CqlSessionExtension.class)
+@CqlSessionSpec(
+    initQueries = {
+      "CREATE TABLE IF NOT EXISTS test (k text, v int, PRIMARY KEY(k, v))",
+    })
+public class ExecuteBatchStreamingTest extends GrpcIntegrationTest {
+
+  @AfterEach
+  public void cleanup(CqlSession session) {
+    session.execute("TRUNCATE TABLE test");
+  }
+
+  @Test
+  public void simpleStreamingBatch(@TestKeyspace CqlIdentifier keyspace) {
+    List<Response> responses = new ArrayList<>();
+
+    StargateStub stub = asyncStubWithCallCredentials();
+    StreamObserver<Response> responseStreamObserver =
+        new StreamObserver<Response>() {
+          @Override
+          public void onNext(Response value) {
+            System.out.println("client onNext: " + value);
+            responses.add(value);
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            System.out.println("error: " + t);
+          }
+
+          @Override
+          public void onCompleted() {
+            System.out.println("client completed ");
+          }
+        };
+
+    StreamObserver<Batch> requestObserver = stub.executeBatchStream(responseStreamObserver);
+
+    requestObserver.onNext(
+        Batch.newBuilder()
+            .addQueries(cqlBatchQuery("INSERT INTO test (k, v) VALUES ('a', 1)"))
+            .addQueries(
+                cqlBatchQuery(
+                    "INSERT INTO test (k, v) VALUES (?, ?)", Values.of("b"), Values.of(2)))
+            .setParameters(batchParameters(keyspace))
+            .build());
+
+    requestObserver.onNext(
+        Batch.newBuilder()
+            .addQueries(
+                cqlBatchQuery(
+                    "INSERT INTO test (k, v) VALUES (?, ?)", Values.of("c"), Values.of(3)))
+            .setParameters(batchParameters(keyspace))
+            .build());
+
+    // all inserted records may be not visible to the subsequent SELECT query
+    // because all reactive calls are non-blocking. Therefore, we need to wait for response of two
+    // batch insert
+    // queries
+    Awaitility.await()
+        .atMost(10, TimeUnit.SECONDS)
+        .until(
+            () -> {
+              return responses.size() == 2;
+            });
+    requestObserver.onCompleted();
+
+    Response response =
+        stubWithCallCredentials()
+            .executeQuery(cqlQuery("SELECT * FROM test", queryParameters(keyspace)));
+    assertThat(response.hasResultSet()).isTrue();
+    ResultSet rs = response.getResultSet();
+    assertThat(new HashSet<>(rs.getRowsList()))
+        .isEqualTo(
+            new HashSet<>(
+                Arrays.asList(
+                    rowOf(Values.of("a"), Values.of(1)),
+                    rowOf(Values.of("b"), Values.of(2)),
+                    rowOf(Values.of("c"), Values.of(3)))));
+  }
+
+  @Test
+  public void streamingQueryWithError(@TestKeyspace CqlIdentifier keyspace) {
```

Comment:
can we have a test case when first batch is OK, and second fails?

Reply:
good catch, I've implemented it and there is a problem with ordering as well, the `onError()` is called so all in-flight requests are canceled. I need to handle it by introducing a new type for streaming (either Result or Error) to not call `onError` for such a case.